unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/EmotesCustomization/EmotesCustomizationComponentController.cs

```diff
@@ -0,0 +1,442 @@
+
+            shortcut2InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut2");
+            shortcut2InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+
+            shortcut3InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut3");
+            shortcut3InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+
+            shortcut4InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut4");
+            shortcut4InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+
+            shortcut5InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut5");
+            shortcut5InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+
+            shortcut6InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut6");
+            shortcut6InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+
+            shortcut7InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut7");
+            shortcut7InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+
+            shortcut8InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut8");
+            shortcut8InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+
+            shortcut9InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut9");
+            shortcut9InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+        }
+
+        internal void OnEquipInputActionTriggered(DCLAction_Hold action)
+        {
+            if (!isEmotesCustomizationSectionOpen || view.selectedCard == null || view.selectedCard.model.isLoading)
+                return;
+
+            if (!view.selectedCard.model.isAssignedInSelectedSlot)
+            {
+                view.EquipEmote(
+                    view.selectedCard.model.id,
+                    view.selectedCard.model.name,
+                    view.selectedSlot);
+            }
+            else
+            {
+                view.UnequipEmote(
+                    view.selectedCard.model.id,
+                    view.selectedSlot);
+            }
+        }
+
+        internal void OnShowInfoInputActionTriggered(DCLAction_Hold action)
+        {
+            if (!isEmotesCustomizationSectionOpen || view.selectedCard == null || view.selectedCard.model.isLoading)
+                return;
+
+            view.OpenEmoteInfoPanel(
+                view.selectedCard.model,
+                view.selectedCard.rarityMark.gameObject.activeSelf ? view.selectedCard.rarityMark.color : Color.grey,
+                view.selectedCard.emoteInfoAnchor);
+        }
+
+        internal void OnNumericShortcutInputActionTriggered(DCLAction_Trigger action)
+        {
+            if (!isEmotesCustomizationSectionOpen || view.selectedCard == null || view.selectedCard.model.isLoading)
+                return;
+
+            switch (action)
+            {
+                case DCLAction_Trigger.ToggleShortcut0:
```

Comment:
I think that's not a good idea here because this method (`OnNumericShortcutInputActionTriggered(DCLAction_Trigger action)`) will always be called to trigger just 1 Emote instead of a range of emotes (that would justify traversing more than 1 with a for).

In any case I think a better alternative would be to just use this line instead of the big switch:
`view.EquipEmote(view.selectedCard.model.id, view.selectedCard.model.name, (int)action);`
But that would only work if the DCLAction_Trigger enum values always correspond to their "DCLAction_Trigger.ToggleShortcutXX" as Nico mentioned.

Reply:
To do that we would need to extract, from the the trigger name string, the number (0 to 9) placed in the final position of this name. It would works but I don't like this approach because it make it strongly dependent on the obligation of call these triggers in a very specific way (in this case: <name of the trigger><slot number>). I fwe decide to change these names in the future, the flow will break.